lib/drivers/tandemTslimDriver.js

```diff
@@ -1133,50 +1131,89 @@ module.exports = function (config) {
   };
 
   var buildTimeChangeRecords = function (data, postrecords) {
-    var timeChangeLogs = filterLogEntries([PUMP_LOG_RECORDS.LID_TIME_CHANGED, PUMP_LOG_RECORDS.LID_DATE_CHANGED],
-      data.log_records);
-
-    timeChangeLogs.forEach(function (change) {
-      var change_ts = change.header_ts;
-      //console.log(change);
-      var fromTime = change_ts - sundial.floor(change_ts, 'day', cfg.timezone);
-      var toTime = fromTime;
-      var fromDay = sundial.floor(change_ts, 'day', cfg.timezone); //TODO this doesn't appear to actually work because of UTC issues.
-      var toDay = fromDay;
-      if (change.header_id == PUMP_LOG_RECORDS.LID_DATE_CHANGED.value) {
-        toDay = BASE_TIME + change.date_after * sundial.MIN_TO_MSEC * 60 * 24;
-        fromDay = BASE_TIME + change.date_prior * sundial.MIN_TO_MSEC * 60 * 24;
+    var timeChangeLogs = filterLogEntries(
+      [PUMP_LOG_RECORDS.LID_TIME_CHANGED],
+      data.log_records
+    );
+
+    var dateChangeLogs = filterLogEntries(
+      [PUMP_LOG_RECORDS.LID_DATE_CHANGED],
+      data.log_records
+    );
+
+    for (var i = 0; i < timeChangeLogs.length; ++i) {
+      var tc = timeChangeLogs[i];
+      var tc_base = sundial.floor(tc.rawTimestamp, 'day').valueOf();
+
+      // look for a corresponding date change event
+      var found = false;
+      var index = tc.index;
```

Comment:
You were using the array index rather than the (persistent) Tandem log index for all this @gniezen...I'm curious why? (Also, I changed it, but we can discuss if there's something I'm missing.)